### ISSUE TITLE
TV does not appear in the apple home app.

### ISSUE THREAD

**Issue by lucaobermaier** [konradknitter/homebridge-philips-android-tv#51]
Hey folks,

my TV appears in the Homebridge UI but not in the home app.
What could be the reason.

Best regards

**Comment by oliwor** [comment]
> Hey folks,
> 
> my TV appears in the Homebridge UI but not in the home app.
> What could be the reason.
> 
> Best regards

Same for me :(

**Comment by konradknitter** [comment]
Did you add TV in iOS Home App?

**Comment by lucaobermaier** [comment]
How to add TV? Only can add new devices.

**Comment by konradknitter** [comment]
Home App
+ Add Accessory
+ No code
Should be on list, code is either same as for homebridge / or in logs

**Comment by oliwor** [comment]
> Home App
> 
> * Add Accessory
> * No code
>   Should be on list, code is either same as for homebridge / or in logs

Can't add a new device. No new device appears. Only the spinning wheel of death appears (loading/searching animation).

This is my log:

[28.8.2021, 11:11:26] [PhilipsAndroidTV] [
  {
    ip: '192.168.1.42',
    mac: 'XX:XX:0A:4C:XX:13',
    name: 'TV Wohnzimmer',
    apiUser: 'tYok3TXXXXX',
    apiPass: 'f951a7fa1637d43ba9435ec8beeaee030XXXXXXXXXXXXXXXXXXX'
  },
  {
    ip: '192.168.1.43',
    mac: 'XX:AD:18:XX:E0:XX',
    name: 'TV Schlafzimmer',
    apiUser: 'NLfLoWasXXXXXXX',
    apiPass: 'b575304ac6567428bc9a430652aXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
  }
]
[28.8.2021, 11:11:26] [PhilipsAndroidTV] Adding new accessory: TV Wohnzimmer
[28.8.2021, 11:11:26] [PhilipsAndroidTV] [TV Wohnzimmer] Connecting to TV to fetch current informations
Preparing Advertiser for 'Homebridge 5244 3B57' using bonjour-hap backend!
Setup Payload:
X-HM://0024YUQTNFVPN
Enter this code with your HomeKit app on your iOS device to pair with Homebridge:
                       
    ┌────────────┐     
    │ 905-XX-XXX │     
    └────────────┘     
                       
Starting to advertise 'Homebridge 5244 3B57' using bonjour-hap backend!
[28.8.2021, 11:11:26] Homebridge v1.3.4 (Homebridge 5244) is running on port 51332.
[28.8.2021, 11:11:26] [PhilipsAndroidTV] [TV Wohnzimmer] TV state {"powerstate":"Standby"}

[28.8.2021, 11:11:28] [PhilipsAndroidTV] [TV Wohnzimmer] Applications setup finished

[28.8.2021, 11:11:28] [PhilipsAndroidTV] [TV Wohnzimmer] Channels setup finished
Preparing Advertiser for 'TV Wohnzimmer B0B8' using bonjour-hap backend!
[28.8.2021, 11:11:28] [PhilipsAndroidTV] Adding new accessory: TV Schlafzimmer
[28.8.2021, 11:11:28] [PhilipsAndroidTV] [TV Schlafzimmer] Connecting to TV to fetch current informations
Starting to advertise 'TV Wohnzimmer B0B8' using bonjour-hap backend!
[28.8.2021, 11:11:28] TV Wohnzimmer B0B8 is running on port 46745.
[28.8.2021, 11:11:28] Please add [TV Wohnzimmer B0B8] manually in Home app. Setup Code: 905-XX-XXX
[28.8.2021, 11:11:29] [PhilipsAndroidTV] [TV Schlafzimmer] TV state {"powerstate":"On"}
[28.8.2021, 11:11:29] [PhilipsAndroidTV] [TV Schlafzimmer] Applications setup finished
[28.8.2021, 11:11:29] [PhilipsAndroidTV] [TV Schlafzimmer] Channels setup finished
Preparing Advertiser for 'TV Schlafzimmer 4EB8' using bonjour-hap backend!
Starting to advertise 'TV Schlafzimmer 4EB8' using bonjour-hap backend!
[28.8.2021, 11:11:29] TV Schlafzimmer 4EB8 is running on port 37115.
[28.8.2021, 11:11:29] Please add [TV Schlafzimmer 4EB8] manually in Home app. Setup Code: 905-XX-XXX
[28.8.2021, 11:11:34] [PhilipsAndroidTV] [TV Wohnzimmer] Start responding again.
[28.8.2021, 11:11:34] [PhilipsAndroidTV] [TV Wohnzimmer] Current application has been changed to NA.
[28.8.2021, 11:11:34] [PhilipsAndroidTV] [TV Wohnzimmer] TV Channel has been changed to NOT AVAILABLE.
[28.8.2021, 11:11:34] [PhilipsAndroidTV] [TV Schlafzimmer] Start responding again.
[28.8.2021, 11:11:34] [PhilipsAndroidTV] [TV Schlafzimmer] TV has been turn on.
[28.8.2021, 11:11:34] [PhilipsAndroidTV] [TV Schlafzimmer] TV volume has been changed to 25%.
[28.8.2021, 11:11:34] [PhilipsAndroidTV] [TV Schlafzimmer] Current application has been changed to NA.

If I try to add a new device with 905-XX-XXX (its the code of the whole homebridge) then again: nothing happens.

Can't add a new device manually, no new device appears...

![image](https://user-images.githubusercontent.com/62522318/131213288-989e12bb-b824-48fb-aca6-3bd63d46e4be.jpeg)

**Comment by konradknitter** [comment]
I see no error, and that print:

[28.8.2021, 11:11:29] Please add [TV Schlafzimmer 4EB8] manually in Home app. Setup Code: 905-XX-XXX

Indicates that externalAccessory (TV) is correctly registered to the Homebridge.

**Comment by oliwor** [comment]
> I see no error, and that print:
> 
> [28.8.2021, 11:11:29] Please add [TV Schlafzimmer 4EB8] manually in Home app. Setup Code: 905-XX-XXX
> > Indicates that externalAccessory (TV) is correctly registered to the Homebridge.

Yes, I will be prompted to add the device manually in home app. But I can't. The device does not appear in home app.
Neither automatically, nor manually via setup code. Why?

In homebridge itself the device appears and I am able to turn on/off the TV, no problem. But the goal is to reach the TV in homekit... and thats not possible at the moment.

**Comment by konradknitter** [comment]
After this homebridge print it should be accessible, and discovered. Nothing else happens on plug-in side.

Like that:
![image](https://user-images.githubusercontent.com/13822445/131213550-e13fa094-65e3-43d1-a876-1749eac1222d.jpeg)

Try to turn homebridge on, off again. Reconfigure plug-in. 
Raise question to homebridge, to get more debug data of platform.

**Comment by oliwor** [comment]
> After this homebridge print it should be accessible, and discovered. 

Yes, it should. But it does not. 

I rebooted homebride several times, deinstalled the plugin several times, reconfigured the plugin several times....
I have no problem adding other devices via plugins. I only have problems with this plugin....

**Comment by BETEP-ok** [comment]
I noticed following pattern. 
if I restart HB service (not the PI itself) the TV is accessible just fine, all the functions working including the APPLE remove 

but about 24hrs later- it simply disappears both - from the HB "Accessories" page and also from the accessories list of the HOME app. 
it started happening since the last major update- so almost several months already.

**Comment by JDTm** [comment]
Only an idea: is your TV connected over LAN? wake on lan enabled?

**Comment by oliwor** [comment]
> Only an idea: is your TV connected over LAN? wake on lan enabled?
> 

Yes and yes

**Comment by Novi001** [comment]
Hello oliwor, 

I had the same problem. I had something from a previous installation in the "config" under accessories. After I deleted that, I was able to put the TV back into HomeKit. The new installation with new pairing writes the plugin in "platforms" ...
Maybe it will help with someone else too.

LG
Novi

**Comment by mrtritop** [comment]
I had this issue too. My workaround was to name it as "TV". It finally worked

**Comment by malditavibora** [comment]
I have the same problem, in my case the philips PUS8118 is recognized and paired under homebridge but it does not appear in the home app, I cannot added it manually because it’s not listed. I’ve tried to use TV as name but its not working
